Extensions/TileMap/JsExtension.js

```diff
@@ -0,0 +1,548 @@
+      propertyName,
+      newValue
+    ) {
+      if (propertyName in objectContent) {
+        if (typeof objectContent[propertyName] === 'boolean')
+          objectContent[propertyName] = newValue === '1';
+        else objectContent[propertyName] = newValue;
+        return true;
+      }
+
+      return false;
+    };
+    objectTileMap.getProperties = function(objectContent) {
+      var objectProperties = new gd.MapStringPropertyDescriptor();
+
+      objectProperties.set(
+        'tiledFile',
+        new gd.PropertyDescriptor(objectContent.tiledFile)
+          .setType('resource')
+          .addExtraInfo('json')
+          .setLabel(_('Tiled file'))
+      );
+      objectProperties.set(
+        'tilemapAtlasImage',
+        new gd.PropertyDescriptor(objectContent.tilemapAtlasImage)
+          .setType('resource')
+          .addExtraInfo('image')
+          .setLabel(_('Tilemap atlas image'))
+      );
+      objectProperties.set(
+        'displayMode',
+        new gd.PropertyDescriptor(objectContent.displayMode)
+          .setType('choice')
+          .addExtraInfo('visible')
+          .addExtraInfo('all')
+          .addExtraInfo('index')
+          .setLabel(_('Display mode'))
+      );
+      objectProperties.set(
+        'layerIndex',
+        new gd.PropertyDescriptor(objectContent.layerIndex.toString())
+          .setType('number')
+          .setLabel(_('Layer index'))
+      );
+      objectProperties.set(
+        'animationSpeed',
+        new gd.PropertyDescriptor(objectContent.animationSpeed.toString())
+          .setType('number')
+          .setLabel(_('Animation speed'))
+      );
+
+      return objectProperties;
+    };
+    objectTileMap.setRawJSONContent(
+      JSON.stringify({
+        tiledFile: '',
+        tilemapAtlasImage: '',
+        displayMode: 'visible',
+        layerIndex: 0,
+        animationSpeed: 1,
+      })
+    );
+
+    objectTileMap.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    objectTileMap.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'TileMap',
+        _('TileMap'),
+        _('Displays a tiled file tilemap'),
+        'JsPlatform/Extensions/tile_map32.png',
+        objectTileMap
+      )
+      .setIncludeFile('Extensions/TileMap/tilemapruntimeobject.js')
+      .addIncludeFile(
+        'Extensions/TileMap/tilemapruntimeobject-pixi-renderer.js'
+      )
+      .addIncludeFile(
+        'Extensions/TileMap/pixi-tilemap/dist/pixi-tilemap.umd.js'
+      )
+      .addIncludeFile(
+        'Extensions/TileMap/pixi-tilemap-helper.js'
+      );
+
+    object
+      .addCondition(
+        'IsTiledFile',
```

Comment:
IsXXX is only for booleans.
Here, name your conditions with the name of what is compared/test. I.e: `TiledFile`